tests/test_git_colocated.rs

```diff
@@ -296,6 +297,67 @@ fn test_git_colocated_squash_undo() {
     "###);
 }
 
+// This test is just like the above test but with a `jj git push` inserted
+#[test]
+fn test_git_colocated_squash_push_undo() {
+    let test_env = TestEnvironment::default();
+    let source_path = test_env.env_root().join("source");
+    git2::Repository::init_bare(&source_path).unwrap();
+    let repo_path = test_env.env_root().join("repo");
+    git2::Repository::clone(&source_path.as_os_str().to_string_lossy(), &repo_path).unwrap();
+    test_env.jj_cmd_success(&repo_path, &["init", "--git-repo=."]);
+    test_env.jj_cmd_success(&repo_path, &["ci", "-m=A"]);
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m=B"]);
+    test_env.jj_cmd_success(&repo_path, &["branch", "set", "master"]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output_divergence(&test_env, &repo_path), @r###"
+    @  rlvkpnrzqnoo 2a3078eda7fe B master
+    ◉  qpvuntsmwlqt a86754f975f9 A
+    ◉  zzzzzzzzzzzz 000000000000
+    "###);
+
+    test_env.jj_cmd_success(&repo_path, &["squash", "-m=A+B"]);
+    insta::assert_snapshot!(get_log_output_divergence(&test_env, &repo_path), @r###"
+    @  royxmykxtrkr 83c0d8df2b78
+    ◉  qpvuntsmwlqt 1873a0811bf5 A+B master
+    ◉  zzzzzzzzzzzz 000000000000
+    "###);
+    test_env.jj_cmd_success(&repo_path, &["git", "push", "--all"]);
+    insta::assert_snapshot!(get_log_output_divergence(&test_env, &repo_path), @r###"
+    @  royxmykxtrkr 83c0d8df2b78
+    ◉  qpvuntsmwlqt 1873a0811bf5 A+B master
+    ◉  zzzzzzzzzzzz 000000000000
+    "###);
+    let stdout = get_truncated_op_log(&test_env, &repo_path, 9);
+    insta::assert_snapshot!(stdout, @r###"
+    @  92f3965076e7 test-username@host.example.com 22 years ago, lasted less than a microsecond
```

Comment:
It looks like this will fail in a year. I'll send a PR to update to absolute timestamps.

Reply:
Thank you! :)